src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyNamesTests.cs

```diff
@@ -0,0 +1,387 @@
+
+                    public C1(int firstDrop, object secondDrop)
+                    {
+                        this.FirstProp = firstDrop;
+                        this.SecondProp = secondDrop;
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task CA1071_ClassPropsMatch_NoDiagnostics_CSharp()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+                using System.Text.Json.Serialization;
+
+                public class C1
+                {
+                    public int FirstProp { get; }
+
+                    public object SecondProp { get; }
+
+                    [JsonConstructor]
+                    public C1(int firstProp, object secondProp)
+                    {
+                        this.FirstProp = firstProp;
+                        this.SecondProp = secondProp;
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task CA1071_ClassPropsMatch_NoDiagnostics_Basic()
+        {
+            await VerifyBasicAnalyzerAsync(@"
+                Imports System.Text.Json.Serialization
+
+                Public Class C1
+                    Property firstProp() As Integer
+                    Property secondProp() as Object
+
+                    Public Sub New(firstDrop as Integer, secondDrop as Object)
+                        Me.firstProp = firstDrop
+                        Me.secondProp = secondDrop
+                    End Sub
+                End Class");
+        }
+
+        [Fact]
+        public async Task CA1071_ClassFieldsDoNotMatch_ConstructorParametersShouldMatchFieldNames_CSharp()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+                using System.Text.Json.Serialization;
+
+                public class C1
+                {
+                    public int firstField;
+                    public object secondField;
+
+                    [JsonConstructor]
+                    public C1(int firstIField, object secondIField)
+                    {
+                        this.firstField = firstIField;
+                        this.secondField = secondIField;
+                    }
+                }",
+            CA1071CSharpFieldResultAt(10, 35, "firstIField", "firstField"),
+            CA1071CSharpFieldResultAt(10, 55, "secondIField", "secondField"));
+        }
+
+        [Fact]
+        public async Task CA1071_ClassFieldsDoNotMatch_ConstructorParametersShouldMatchFieldNames_Basic()
+        {
+            await VerifyBasicAnalyzerAsync(@"
+                Imports System.Text.Json.Serialization
+
+                Public Class C1
+                    Public firstField as Integer
+                    Public secondField as Object
+
+                    <JsonConstructor>
+                    Public Sub New(firstIField as Integer, secondIField as Object)
+                        Me.firstField = firstIField
+                        Me.secondField = secondIField
```

Comment:
Consider adding another test without 'Me' to ensure proper codefix behavior when it's implemented.

Reply:
Is it something that can happen? I haven't written a code fix yet, but I certainly will add another test if it something that can go wrong.